components/site/src/lib.rs

```diff
@@ -977,8 +970,6 @@ impl Site {
 
     /// What it says on the tin
     pub fn render_sitemap(&self) -> Result<()> {
-        ensure_directory_exists(&self.output_path)?;
```

Comment:
I'd rather keep those around even if in theory it's already created elsewhere

Reply:
The reason why I removed it here is because `write_content` always creates directories when called, so, you can rely on that instead of creating them ahead of time.

Reply:
It's ok in the grand scheme of things but creates inter-dependencies in case we decide to not use write_content for some reason. Maybe it's fine.

Reply:
That's fair. My thought process was that we should delay creating the directory as long as possible to avoid making them unnecessarily, but I'm open to adding back in checks that are guarded on whether things will actually be written.